llvm/lib/Target/AMDGPU/SIMachineFunctionInfo.cpp

```diff
@@ -195,6 +195,9 @@ SIMachineFunctionInfo::SIMachineFunctionInfo(const Function &F,
     VGPRForAGPRCopy =
         AMDGPU::VGPR_32RegClass.getRegister(ST.getMaxNumVGPRs(F) - 1);
   }
+
+  if (F.hasFnAttribute("amdgpu-disable-tbuffer-combine"))
+    setDisableTBufferCombine(true);
```

Comment:
I think Matt's suggestion was that the conservative behaviour should be the default, so the tbuffer merging should be disabled by default, and only enabled when an attribute is set.

Reply:
I see where you're coming from on this, and I had similar thoughts in https://github.com/llvm/llvm-project/pull/115479. But being always correct by default has some clear advantages. The price to pay for this is to always set a certain flag to enable relaxed mode.